partiql-lang/src/test/kotlin/org/partiql/lang/planner/transforms/PartiQLSchemaInferencerTests.kt

```diff
@@ -1072,6 +1072,1179 @@ class PartiQLSchemaInferencerTests {
+                    )
+                )
+            ),
+            // EXCLUDE regression test (behavior subject to change pending RFC); could give error/warning
+            SuccessTestCase(
+                name = "invalid exclude tuple attr",
+                query = """SELECT * EXCLUDE t.a.b
+                    FROM <<
+                        {
+                            'a': [
+                                { 'b': 0 },
+                                { 'b': 1 },
+                                { 'b': 2 }
+                            ]
+                        }
+                    >> AS t""",
+                expected = BagType(
+                    elementType = StructType(
+                        fields = mapOf(
+                            "a" to ListType(
+                                elementType = StructType(
+                                    fields = mapOf(
+                                        "b" to StaticType.INT
+                                    ),
+                                    contentClosed = true,
+                                    constraints = setOf(TupleConstraint.Open(false), TupleConstraint.UniqueAttrs(true))
+                                )
+                            )
+                        ),
+                        contentClosed = true,
+                        constraints = setOf(TupleConstraint.Open(false), TupleConstraint.UniqueAttrs(true), TupleConstraint.Ordered)
+                    )
+                )
+            ),
+            // EXCLUDE regression test (behavior subject to change pending RFC); could give error/warning
+            SuccessTestCase(
+                name = "invalid exclude tuple wildcard",
+                query = """SELECT * EXCLUDE t.a.*
+                    FROM <<
+                        {
+                            'a': [
+                                { 'b': 0 },
+                                { 'b': 1 },
+                                { 'b': 2 }
+                            ]
+                        }
+                    >> AS t""",
+                expected = BagType(
+                    elementType = StructType(
+                        fields = mapOf(
+                            "a" to ListType(
+                                elementType = StructType(
+                                    fields = mapOf(
+                                        "b" to StaticType.INT
+                                    ),
+                                    contentClosed = true,
+                                    constraints = setOf(TupleConstraint.Open(false), TupleConstraint.UniqueAttrs(true))
+                                )
+                            )
+                        ),
+                        contentClosed = true,
+                        constraints = setOf(TupleConstraint.Open(false), TupleConstraint.UniqueAttrs(true), TupleConstraint.Ordered)
+                    )
+                )
+            ),
```

Comment:
A couple more test cases that can be beneficial: 
```
-- demonstrate expression in exclude don't not lead to typing error.  
SELECT * EXCLUDE t.b
                    FROM <<
                        {
                            'a': <<
                                { 'b': 0 },
                                { 'b': 1 },
                                { 'b': 2 }
                            >>
                        }
                    >> AS t
                    
-- demonstrate expression in exclude don't not lead to typing error.  
SELECT * EXCLUDE nonsense.a
                    FROM <<
                        {
                            'a': <<
                                { 'b': 0 },
                                { 'b': 1 },
                                { 'b': 2 }
                            >>
                        }
                    >> AS t 
                            
-- I am assuming
-- bag ( struct ( list ( unionOf( struct(b: int, c: unionOf(missing, int) ), struct(b: int, c: unionOf(missing, NULL, MISSSING) ), struct(b: int, c: unionOf(missing, decimal) ))) ) ) 
SELECT * EXCLUDE t.a[0].c
                    FROM <<
                        {
                            'a': [
                                { 'b': 0,
                                  'c': 0
                                },
                                { 'b': 1,
                                  'c': NULL
                                },
                                { 'b': 2,
                                  'c': 0.1
                                 }
                            ]
                     
                        } 
                    >> AS t                               
```

Reply:
Yes to your assumption. I will add those test cases.